plugin/src/main/java/io/jenkins/plugins/casc/SchemaGeneration.java

```diff
@@ -155,20 +148,21 @@ else if (configuratorObject instanceof HeteroDescribableConfigurator) {
                 if (implementorsMap.size() != 0) {
                     Iterator<Map.Entry<String, Class>> itr = implementorsMap.entrySet().iterator();
 
-                    JSONObject implementorObject = new JSONObject();
-                    implementorObject.put("type", "object");
+                    JSONArray oneOfJsonArray = new JSONArray();
+                    JSONObject finalHetroConfiguratorObject = new JSONObject();
                     while (itr.hasNext()) {
                         Map.Entry<String, Class> entry = itr.next();
+                        JSONObject implementorObject = new JSONObject();
                         implementorObject.put("properties",
                             new JSONObject().put(entry.getKey(),
-                            new JSONObject().put("$ref","#/definitions/" + entry.getValue())));
+                            new JSONObject().put("$ref","#/definitions/" + entry.getValue())).toString());
```

Comment:
@casz The builder seems to be adding a strang backslash I am not sure if it is supposed to be added
` "properties": "{\"list\":{\"$ref\":\"#/definitions/class hudson.model.ListView\"}}`